File: backend/src/entities/AIConversation.ts
import { Entity, PrimaryGeneratedColumn, Column } from "typeorm";

export type AIConversationStatus = "active" | "archived";

/**
 * One chat thread between a user and the FabriXAI-backed Form Builder
 * assistant (see `aiAssistantService.ts`). `formId` links the conversation to
 * whichever campaign/form editor it was opened from — null for a
 * conversation started without an open form (e.g. a general "find my
 * previous campaigns" chat from a list page).
 */
@Entity("AIConversations")
export class AIConversation {
  @PrimaryGeneratedColumn("uuid")
  id!: string;

  @Column({ type: "uniqueidentifier" })
  userId!: string;

  @Column({ type: "uniqueidentifier", nullable: true })
  formId!: string | null;

  @Column({ type: "nvarchar", length: 200, nullable: true })
  title!: string | null;

  @Column({ type: "nvarchar", length: 20, default: "active" })
  status!: AIConversationStatus;

  @Column({ type: "datetimeoffset", default: () => "SYSDATETIMEOFFSET()" })
  createdAt!: Date;

  @Column({ type: "datetimeoffset", default: () => "SYSDATETIMEOFFSET()" })
  updatedAt!: Date;
}

File: backend-py/app/models/ai_conversation.py
"""Port of `backend/src/entities/AIConversation.ts`."""

from __future__ import annotations

from typing import Literal, Optional

from sqlalchemy.orm import Mapped

from app.models.base import Base, datetimeoffset_now, nvarchar, uuid_col, uuid_pk

AIConversationStatus = Literal["active", "archived"]


class AIConversation(Base):
    __tablename__ = "AIConversations"

    id: Mapped[str] = uuid_pk()
    userId: Mapped[str] = uuid_col()
    formId: Mapped[Optional[str]] = uuid_col(nullable=True)
    title: Mapped[Optional[str]] = nvarchar(200, nullable=True)
    status: Mapped[AIConversationStatus] = nvarchar(20, default="active")
    createdAt = datetimeoffset_now()
    updatedAt = datetimeoffset_now()
